src/core/ext/client_channel/client_channel.c

```diff
@@ -364,6 +365,8 @@ static void on_resolver_result_changed(grpc_exec_ctx *exec_ctx, void *arg,
   if (old_lb_policy != NULL) {
     grpc_pollset_set_del_pollset_set(
         exec_ctx, old_lb_policy->interested_parties, chand->interested_parties);
+    GRPC_LB_POLICY_WEAK_UNREF(exec_ctx, old_lb_policy,
```

Comment:
Just to verify my understanding here, this works because on_resolver_result_changed() always gets called with chand->resolver_result == NULL before the client channel is destroyed, right?